src/v0/destinations/pinterest_tag/transform.js

```diff
@@ -119,25 +137,45 @@ const process = (event) => {
   }
 
   toSendEvents.forEach((sendEvent) => {
-    respList.push(responseBuilderSimple(sendEvent));
+    respList.push(responseBuilderSimple(sendEvent, destination));
   });
   return respList;
 };
 
-const generateBatchedPayloadForArray = (events) => {
+const generateBatchedPayloadForArray = (events, { Config }) => {
+  const { apiVersion } = Config;
   const { batchedRequest } = defaultBatchRequestConfig();
   const batchResponseList = events.map((event) => event.body.JSON);
 
-  batchedRequest.body.JSON = {
-    data: batchResponseList,
-  };
+  batchedRequest.body.JSON = { data: batchResponseList };
   batchedRequest.endpoint = ENDPOINT;
-  batchedRequest.headers = {
-    'Content-Type': 'application/json',
-  };
+  batchedRequest.headers = { 'Content-Type': 'application/json' };
+
+  if (apiVersion === API_VERSION.v5) {
+    const { adAccountId, conversionToken } = Config;
+    const endpoint = getV5EventsEndpoint(adAccountId);
+    batchedRequest.endpoint = endpoint;
+    batchedRequest.headers = {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${conversionToken}`,
+    };
+  }
```

Comment:
Non-Blocking- Do we need it here? As `batchEvents` will get called after transformation so we will have the specific endpoints, isn't it?

Reply:
upadted